### ISSUE TITLE
PostGisGeometryType is not detecting Z ordinate for polygons

### ISSUE THREAD

**Issue by eduardotsj** [nhibernate/NHibernate.Spatial#120]
On PostGisGeometryType (line 73) is using geometry.InteriorPoint to determine whether the Z-axis is been used or not. 
It happens that even with all coordinates setting the Z value, the InteriorPoint is always returning 2D for polygons. Not sure if I'm missing something, but all my geometries are been saved without the Z-axis
I'm using NHibernate.Spatial.PostGis 5.3.0 with dotnet core 3.1.
I've made a change to use the first coordinate of the geometry instead and then it worked great:

This is my proposed solution:

            var basePoint = geometry.Coordinate;
            if (basePoint != null && !double.IsNaN(basePoint.Z))
            {
                ordinates |= Ordinates.Z;
            }
            // GeoAPI 1.7.5 does not support M ordinate

Does anyone see any issue with that?

**Comment by peetw** [comment]
Hi @eduardotsj thanks for reporting this. I have been a bit swamped recently, but will try and take a look at it this week.

**Comment by peetw** [comment]
Proposed solution looks to make sense. Do you have an example polygon that causes the original issue that I can test with?

**Comment by eduardotsj** [comment]
Unfortunately I can't send you a functional code, but at least I can tell you that this is the geometry that I was saving:

entity.Geom = new Polygon(new LinearRing(new Coordinate[] { new Coordinate(0, 0, 1), new Coordinate(0, 100, 2), new Coordinate(100, 100, 3), new Coordinate(100, 0, 4), new Coordinate(0, 0, 5) }));

When I read it back, the Z-values are gone.
I've forked this version and applied the solution I've proposed. I'm running it for a while without any issues.
Of course the spectrum of my use cases are limited.
I also tried to follow the logic of InteriorPoint function to see if it was easier to make it to include the Z-axis, but it is out of this project.
